tikv-client/src/main/java/com/pingcap/tikv/columnar/TiColumnVector.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pingcap.tikv.columnar;
+
+import com.pingcap.tikv.types.DataType;
+import java.math.BigDecimal;
+
+/**
+ * An interface is mostly copied from Spark's ColumnVector (we do not link it here because we do not
+ * want to pollute tikv java client's dependencies).
+ *
+ * <p>Most of the APIs take the rowId as a parameter. This is the batch local 0-based row id for
+ * values in this TiColumnVector.
+ *
+ * <p>Spark only calls specific `get` method according to the data type of this {@link
+ * TiColumnVector}, e.g. if it's int type, Spark is guaranteed to only call {@link #getInt(int)} or
+ * {@link #getInts(int, int)}.
+ *
+ * <p>TiColumnVector is expected to be reused during the entire data loading process, to avoid
+ * allocating memory again and again.
+ */
+public abstract class TiColumnVector implements AutoCloseable {
```

Comment:
question: why is TiColumnVector used as AutoClosable? it doesn't seem necessary.

Reply:
Later, we will use off heap implementation.